### PR TITLE
Fixes #507 -- added download with original filename to EIO admin

### LINKED ISSUE #507: As admin, I want to download a document, in the admin, by its filename
... so I can open it directly.

**Description**
Currently, the're a "inhoud" field that let's you download the contents of the Document/Informatieobject in the admin. This file is by default a `.bin` file.

Most of the time, a filename (bestandsnaam) is part of the metadata. If that's the case, I want a link that let's me download the file with it's proper filename. This can be a new line/field in the admin.

PS. This wish was a result of the Delft POC.

### PR DESCRIPTION
**Pull request by stevenbal**
issue: https://github.com/open-zaak/open-zaak/issues/507

### CODE REVIEW

File: src/openzaak/components/documenten/admin.py [hunk 1]
```diff
@@ -23,6 +26,22 @@
 )
 
 
+class FileDownloadMixin:
```
Comment (sergei-maertens) [inline src/openzaak/components/documenten/admin.py]: Please don't make _everything_ a mixin. There is only one place where this is used now - this should really just be a method on the `ModelAdmin` where it's used.

File: src/openzaak/components/documenten/admin.py [hunk 2]
```diff
@@ -251,3 +273,22 @@ def get_object_actions(self, obj):
             link_to_related_objects(Gebruiksrechten, obj.canonical),
             link_to_related_objects(ObjectInformatieObject, obj.canonical),
         )
+
+    def get_urls(self):
```
Comment (sergei-maertens) [inline src/openzaak/components/documenten/admin.py]: why are you not re-using the URL(s) from django-privates that are already provided?

File: src/openzaak/components/documenten/admin.py [hunk 3]
```diff
@@ -251,3 +273,22 @@ def get_object_actions(self, obj):
...
+                "<int:pk>/download/",
+                self.download,
+                name="documenten_enkelvoudiginformatieobject_download",
+            ),
+        ]
+        return my_urls + urls
+
+    def download(self, request, pk):
+        obj = self.get_object(request, pk)
+        response = FileResponse(obj.inhoud)
```
Comment (sergei-maertens) [inline src/openzaak/components/documenten/admin.py]: You _really_ don't want to have Django serving potentially massive files from disk. This should use `sendfile` on nginx, all of this is implemented in django-privates already. Please look up the documentation/implementation of django-privates on how to customize the attachment filename.

File: src/openzaak/components/documenten/admin.py (hunk 1 above)
Comment (stevenbal) [reply src/openzaak/components/documenten/admin.py]: I figured since I added it to both the `EnkelvoudigInformatieObjectInline` and the `EnkelvoudigInformatieObjectAdmin`

Comment (sergei-maertens) [reply src/openzaak/components/documenten/admin.py]: my bed, I missed that in the diff then :) 😅

File: src/openzaak/components/documenten/admin.py [hunk 4]
```diff
@@ -251,3 +253,23 @@ def get_object_actions(self, obj):
...
+
+    def formfield_for_dbfield(self, db_field, request, **kwargs):
+        field = super().formfield_for_dbfield(db_field, request, **kwargs)
+        private_media_fields = self.get_private_media_fields()
+        if db_field.name in private_media_fields:
+            view_name = self._get_private_media_view_name(db_field.name)
+            obj = self.get_object(request, request.resolver_match.kwargs["object_id"])
+            field.widget = PrivateFileWidget(
+                url_name="admin:%s" % view_name,
+                attrs={"display_value": obj.bestandsnaam},
```
Comment (stevenbal) [inline src/openzaak/components/documenten/admin.py]: @sergei-maertens there has to be a better way to change the display value for the download link of the file in the admin right? I want it to show the `bestandsnaam` instead of the path where it is stored in private_media

File: src/openzaak/components/documenten/tests/admin/test_eio_download.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        bestand_field = response.html("div", {"class": "field-inhoud"})[0]
+        download_link = bestand_field("a")[0]
+
+        self.assertEqual(download_link.text, "iets.txt")
+
+        download_response = self.app.get(download_link.attrs["href"])
+        self.assertEqual(download_response.content, b"STUFF")
+        self.assertEqual(
+            download_response.content_disposition, 'attachment; filename="iets.txt"'
+        )
```
Comment (joeribekker) [inline src/openzaak/components/documenten/tests/admin/test_eio_download.py]: Can you add a test what the filename is when "bestandsnaam" is empty?
